apps/golf-gps/golf-gps.js

```diff
@@ -0,0 +1,306 @@
+    46,
+    atob("FBkbFBsaHBobGxsbEw=="),
+    48 | 65536
+  );
+};
+
+var courseData = require("Storage").open("course-data", "r");
+
+var lastFix = {
+  lat: 0,
+  lon: 0,
+  alt: 0, // altitude in m
+  speed: 0, // km/h
+  course: 0, // heading in degrees
+  time: 0,
+  satellites: 0,
+  fix: 0,
+  hdop: 0, // x5 ~ meter accuracy
+};
+
+const zeroPad = (num, places) => String(num).padStart(places, '0');
+
+function onGPS(fix) {
+  Object.assign(lastFix, fix);
+  if (lastFix.fix && playON) showPlayData();
+}
+
+function radians(degrees) {
+  return degrees * Math.PI / 180;
+}
+
+function readOneCourseData() {
+  // Clear previous data
+  lat = [];
+  lon = [];
+  par = [];
+  let l = courseData.readLine();
+  courseName = l;
+  // Read one course data & initialize score
+  for (let i = 1; i < 19; i++) {
+    l = courseData.readLine();
+    if (l !== undefined) {
+      const parts = l.split(',');
+      lat[i] = parseFloat(parts[0]);
+      lon[i] = parseFloat(parts[1]);
+      par[i] = parseInt(parts[2]);
+      score[i] = 0;
+    }
+  }
+}
+
+function distanceCalc(lat1, long1, lat2, long2) {
+  let delLat = Math.abs(lat1 - lat2) * 111194.9; // 111194.9 = (2*6371000*pi)/360, 6371000 ~ Earth's average radius
+  let delLong = Math.abs(long1 - long2) * 111194.9 * Math.cos(radians((lat1+lat2)/2));
+  return Math.sqrt(delLat * delLat + delLong * delLong)/0.9144; // in yards
+}
+
+function mainMenu() {
+  E.showPrompt("Play now or\n\nView scores?", {
+    title: "Golf GPS",
+    buttons: {
+      "PLAY": 1,
+      "VIEW": 2
+    }
+  }).then(choice => {
+    if (choice === 1) fixGPS();
+    else browseScore();
+  });
+}
+
+function browseScore() {
+  const scoreFiles = require("Storage").list(/^Scorecard-/);
+  if (scoreFiles.length === 0) {
+    E.showPrompt("No score file found.\n\nYou need to play at least a game.", {
+      title: `Error`,
+      buttons: {
+        "END": 1
+      }
+    }).then(choice => {
+      if (choice === 1) load();
+    });
+  }
+  let fileIndx = scoreFiles.length - 1;
+
+  function browseFiles() {
+    const browsefile = require("Storage").open(scoreFiles[fileIndx].substring(0, 18), "r");
+    const l = browsefile.read(80);
+
+    E.showPrompt(l, {
+      buttons: {
+        "<<": 1,
+        ">>": 2,
+        "End": 3
+      }
+    }).then(choice => {
+      if (choice === 1) fileIndx = (fileIndx - 1 + scoreFiles.length) % scoreFiles.length;
+      else if (choice === 2) fileIndx = (fileIndx + 1) % scoreFiles.length;
+      else if (choice === 3) load();
```

Comment:
I'm happy if you prefer this - if you do like though, you can use a switch:
```javascript
switch(choice) {
  case 1:
    fileIndx = (fileIndx - 1 + scoreFiles.length) % scoreFiles.length;
    break;
  case 2:
    fileIndx = (fileIndx + 1) % scoreFiles.length;
    break;
  case 3:
    load();
    break;
}
```

Reply:
It works either way, but I like your suggested 'switch' better. Thanks!

Reply:
My pleasure!